pepsico/comparison.py

```diff
@@ -1,14 +1,66 @@
 import xarray as xr
-    
-# defining the path to file  
-ds_zarr = '../data/remic/mydatafiles/zarr/test/'
+import os
+#from pathlib import Path 
 
-ds_nc = "/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted/global/daily/historical/MPI-ESM1-2-HR/mpi-esm1-2-hr_r1i1p1f1_w5e5_historical_hurs_global_daily_1951_1960.nc" 
-  
-# using the Dataset() function  
-zarr_dSet = xr.open_zarr(ds_zarr)
+# defining the path to file
+ds_zarr = '/data/remic/mydatafiles/zarr/test/hurs'
 
-nc_dSet = xr.open_dataset(ds_nc)
+ds_nc_dir = '/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted/global/daily/historical/MPI-ESM1-2-HR/' #mpi-esm1-2-hr_r1i1p1f1_w5e5_historical_hurs_global_daily_1951_1960.nc'
+#iterating over all the years
+
+#only using NetCDF files that contain 'hurs' variable
+def contains_hurs(filename):
+    return 'hurs' in filename
+
+#function that compares nc files(hurs) vs zarred file
+def compare_datasets(nc_file_path, ds_zarr):
+    #returns true if datasets are identical, false otherwise
+
+    #open datasets
+    zarr_dSet = xr.open_zarr(ds_zarr)
+    nc_dSet = xr.open_dataset(nc_file_path)[['hurs']]
+
+    #mapping from zar to nc dimensions (X,Y,T) vs (lon,lat,time)
+    coord_map = {'X': 'lon', 'Y': 'lat', 'T': 'time'}
+
+    #xararay rename function to reset zarr dimensions to nc labels 
+    zarr_dSet = zarr_dSet.rename({orig_dim: new_dim for orig_dim, new_dim in coord_map.items() if orig_dim in zarr_dSet.dims})
+    zarr_dSet = zarr_dSet.rename({orig_coord: new_coord for orig_coord, new_coord in coord_map.items() if orig_coord in zarr_dSet.coords})
```

Comment:
it doesn't matter here, but in the future, let's have our libraries use our conventions, that is to say X, Y, T and thus change 3rd party conventions to ours

Reply:
Got it, I changed it for future reference.